cli/src/pcluster/models/s3_bucket.py

```diff
@@ -354,6 +443,13 @@ def _check_default_bucket(cls, service_name: str, artifact_directory: str, stack
             bucket.check_bucket_is_bootstrapped()
         except AWSClientError as e:
             cls._configure_bucket(bucket, e)
+        else:
+            # Bucket exists and is bootstrapped, ensure bucket policy is up-to-date
+            try:
+                bucket.ensure_bucket_policy_for_logs()
```

Comment:
Improvement, that can be done in a follow up PR.

We should configure the bucket policy within `_configure_bucket()` where we are already setting the DenyHTTP statement https://github.com/aws/aws-parallelcluster/blob/8ac2c636f7941bc8983ee75f6abf202028b2d031/cli/src/pcluster/models/s3_bucket.py#L144-L156

At that point we must change the logic to check if the bucket is already bootstrapped:
1. change the bootstrap file body [here](https://github.com/aws/aws-parallelcluster/blob/8ac2c636f7941bc8983ee75f6abf202028b2d031/cli/src/pcluster/models/s3_bucket.py#L191) to reflect what are the bootstrapped features required, e.g. basic (what we already have up to Pc 3.11.1) and export-logs (introduced in PC 3.12.0). 
2. if the bootstrap file does not exist or it does not contain all the required features, then overwrite the bucket policy

Reply:
With the changes removing SIDs check, we will need this logic in this PR. Need some more time to implement

Reply:
Done.